agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/preaggregatedmetrics/RequestCustomDimensionsExtractor.java

```diff
@@ -0,0 +1,80 @@
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.azure.monitor.opentelemetry.exporter.implementation.preaggregatedmetrics;
+
+import com.azure.monitor.opentelemetry.exporter.implementation.builders.AbstractTelemetryBuilder;
+import com.azure.monitor.opentelemetry.exporter.implementation.models.ContextTagKeys;
+
+public final class RequestCustomDimensionsExtractor {
+
+  // visible for testing
+  public static final String MS_METRIC_ID = "_MS.metricId";
+  public static final String REQUEST_METRIC_ID = "requests/duration";
+  public static final String MS_IS_AUTOCOLLECTED = "_MS.IsAutocollected";
+  public static final String TRUE = "True";
+  public static final String FALSE = "False";
+  public static final String MS_PROCESSED_BY_METRIC_EXTRACTORS = "_MS.ProcessedByMetricExtractors";
+  public static final String PERFORMANCE_BUCKET = "request/performanceBucket";
+  public static final String REQUEST_RESULT_CODE = "request/resultCode";
+  public static final String OPERATION_SYNTHETIC = "operation/synthetic";
+  public static final String CLOUD_ROLE_NAME = "cloud/roleName";
+  public static final String CLOUD_ROLE_INSTANCE = "cloud/roleInstance";
+  public static final String REQUEST_SUCCESS = "request/success";
+
+  public static void updatePreAggMetricsCustomDimensions(
+      AbstractTelemetryBuilder metricTelemetryBuilder,
+      String perfBucket,
+      Long statusCode,
+      boolean success) {
+    metricTelemetryBuilder.addProperty(MS_METRIC_ID, REQUEST_METRIC_ID);
+    metricTelemetryBuilder.addProperty(MS_IS_AUTOCOLLECTED, TRUE);
+    // this flag will inform the ingestion service to stop post-aggregation
+    metricTelemetryBuilder.addProperty(MS_PROCESSED_BY_METRIC_EXTRACTORS, TRUE);
+
+    metricTelemetryBuilder.addProperty(PERFORMANCE_BUCKET, perfBucket);
+    // TODO how to assign "request/result_code" to RPC duration metrics?
```

Comment:
i didn't add "request/result_code" to rpc pre-aggregated metrics since there is no status_code available from the otel.